src/Orleans.Core/Providers/IOrleansProvider.cs

```diff
@@ -11,7 +11,6 @@ namespace Orleans.Providers
     /// <summary>
     /// Base interface for all type-specific provider interfaces in Orleans
     /// </summary>
```

Comment:
Unclear, but I think it will be removed something like:
1) Add legacy support for providers (this PR)
2) Port all supported providers directly to names lifecycle participants that do not inherit from IProvider.
3) Move IProvider to legacy packages (Core.Legacy)
Somewhere along the way, we'll need to move cluster and client configuratiosn to legacy as well.